src/em/prov/em_provisioning.cpp

```diff
@@ -661,6 +662,156 @@ cJSON *em_provisioning_t::create_enrollee_bsta_list(ec_connection_context_t *con
+        cJSON_Delete(bsta_configuration_object);
+    }
+
+    cJSON *discovery_object = cJSON_CreateObject();
+    if (discovery_object == nullptr) {
+        printf("%s:%d: Failed to create Discovery Object for DPP Configuration Object\n", __func__, __LINE__);
+        cJSON_Delete(bsta_configuration_object);
+    }
+    const em_network_ssid_info_t* network_ssid_info = dm->get_network_ssid_info_by_haul_type(em_haul_type_backhaul);
+    ASSERT_NOT_NULL(network_ssid_info, nullptr, "%s:%d: No backhaul BSS found, cannot create bSTA Configuration Object\n", __func__, __LINE__);
+    if (!cJSON_AddStringToObject(discovery_object, "SSID", network_ssid_info->ssid)) {
+        printf("%s:%d: Could not add \"SSID\" to bSTA Configuration Object\n", __func__, __LINE__);
+        cJSON_Delete(bsta_configuration_object);
+        cJSON_Delete(discovery_object);
+        return nullptr;
+    }
+    // Find "BSSID" and "RUID" since not contained in `em_network_ssid_info_t`
+    for (unsigned int i = 0; i < dm->get_num_bss(); i++) {
+        const dm_bss_t *bss = dm->get_bss(i);
+        if (!bss) continue;
+        if (bss->m_bss_info.backhaul_use && strncmp(bss->m_bss_info.ssid, network_ssid_info->ssid, strlen(network_ssid_info->ssid)) == 0) {
+            if (!cJSON_AddStringToObject(discovery_object, "BSSID", reinterpret_cast<const char *>(bss->m_bss_info.id.bssid))) {
+                printf("%s:%d: Failed to add \"BSSID\" to bSTA Configuration Object\n", __func__, __LINE__);
+                cJSON_Delete(bsta_configuration_object);
+                cJSON_Delete(discovery_object);
+                return nullptr;
+            }
+            if (!cJSON_AddStringToObject(discovery_object, "RUID", reinterpret_cast<const char *>(bss->m_bss_info.id.ruid))) {
+                printf("%s:%d: Failed to add \"RUID\" to bSTA Configuration Object\n", __func__, __LINE__);
+                cJSON_Delete(bsta_configuration_object);
+                cJSON_Delete(discovery_object);
+                return nullptr;
+            }
+        }
+    }
+
+    cJSON *credential_object = cJSON_CreateObject();
+    if (credential_object == nullptr) {
+        printf("%s:%d: Failed to create credential object for DPP Configuration object.\n", __func__, __LINE__);
+        cJSON_Delete(discovery_object);
+        cJSON_Delete(bsta_configuration_object);
+        return nullptr;
+    }
+
+    std::string akm_suites = {};
+    bool needs_psk_hex = false;
+    for (unsigned int i = 0; i < network_ssid_info->num_akms; i++) {
+        if (!akm_suites.empty()) akm_suites += "+";
+        akm_suites += util::akm_to_oui(network_ssid_info->akm[i]);
+
+    }
+    // "psk_hex" is a conditional field,
+    // present only if PSK or AKM or SAE is a selected AKM
+    const auto check_needs_psk_hex = [](std::string akm) -> bool {
+        // psk || sae
+        return akm == "000FAC02"
+        || akm == "000FAC08";
+    };
```

Comment:
While a bit slower, can you possibly do this?
```suggestion
    const auto check_needs_psk_hex = [](std::string akm) -> bool {
        // psk || sae
        return akm == util::akm_to_oui("psk")
        || akm == util::akm_to_oui("sae")
    };
```
to enhance readability? It's not needed though